File: java/src/main/java/io/fivexer/sdk/model/DecisionCandidate.java
package io.fivexer.sdk.model;

import java.util.Map;

/**
 * One evaluated candidate within a {@link Decision}. {@code workerId} is pulled out;
 * every other field (score, eligible, chosen, veto reasons, …) is kept verbatim in
 * {@link #getDetail()} — reason kinds are additive over time, so unknown keys are
 * forward-compatible, not errors.
 */
public class DecisionCandidate {
    private String workerId;
    private Map<String, Object> detail;

    public String getWorkerId() { return workerId; }
    public Map<String, Object> getDetail() { return detail; }

    public void setWorkerId(String workerId) { this.workerId = workerId; }
    public void setDetail(Map<String, Object> detail) { this.detail = detail; }
}

File: php/src/Model/DecisionCandidate.php
<?php

declare(strict_types=1);

namespace Fivexer\SDK\Model;

/**
 * One evaluated candidate within a Decision. workerId is pulled out; every other
 * field (score, eligible, chosen, veto reasons, ...) is kept verbatim in detail -
 * reason kinds are additive over time, so unknown keys are forward-compatible.
 */
final class DecisionCandidate
{
    /**
     * @param array<string, mixed> $detail
     */
    public function __construct(
        public readonly ?string $workerId,
        public readonly array $detail,
    ) {
    }

    /** @param array<string, mixed> $data */
    public static function fromArray(array $data): self
    {
        $workerId = isset($data['workerId']) ? (string) $data['workerId'] : null;
        $detail = $data;
        unset($detail['workerId']);
        return new self($workerId, $detail);
    }
}
